internal/printer/printer.go

```diff
@@ -0,0 +1,41 @@
+package printer
+
+import (
+	"encoding/json"
+
+	"gopkg.in/yaml.v3"
+)
+
+type Output string
+
+const (
+	JSON Output = "json"
+	YAML Output = "yaml"
+)
+
+func PrintOutput(output Output, p any) (string, error) {
```

Comment:
In terms of code maintenance, I'd like to see this abstracted more and configured as part of startup. So something early on from the root command could read the output flag and set a global config somewhere. That way its not repeated in individual commands and those commands just call a basic "print" type function that knows what format. 

But another thing to refactor for now